docs/custom_parsers.md

```diff
@@ -0,0 +1,221 @@
+(custom-parsers)=
+
+# Custom parsers
+
+This page explains how to write a custom parser for VirtualiZarr, to extract chunk references from an archival data format not already supported by the main package.
+This is advanced material intended for 3rd-party developers, and assumes you have read the page on [Data Structures](data_structures.md).
+
+```{note}
+"Parsers" were previously known variously as "readers" or "backends" in older versions of VirtualiZarr. We renamed them to avoid confusion with obstore readers and xarray backends.
+```
+
+## What is a VirtualiZarr parser?
+
+All VirtualiZarr parsers are simply callables that accept a path to a file of a specific format and an instantiated [obstore](https://developmentseed.org/obstore/latest/) store to read data from it with, and return an instance of the :py:class:~`virtualizarr.manifests.ManifestStore` class containing information about the contents of that file.
+
+```python
+from virtualizarr.manifests import ManifestStore, ObjectStoreRegistry
+
+
+def custom_parser(file_url: str, object_store: ObjectStore) -> ManifestStore:
+    # access the file's contents, e.g. using the ObjectStore instance
+    readable_file = obstore.open_reader(object_store, file_url)
+
+    # parse the file contents to extract its metadata
+    # this is generally where the format-specific logic lives
+    manifestgroup: ManifestGroup = extract_metadata(readable_file)
+
+    # optionally create an object store registry, used to actually load chunk data from file later
+    registry = ObjectStoreRegistry({store_prefix: object_store})
+
+    # construct the Manifeststore from the parsed metadata and the object store registry
+    return ManifestStore(group=manifestgroup, store_registry=registry)
```

Comment:
Writing this out made me realize it's a bit weird that an `ObjectStore` is required by the call signature, but not actually technically needed by the code...